File: php/features.php
<?php
declare(strict_types=1);

// GithubApi2 SDK feature factory

require_once __DIR__ . '/feature/BaseFeature.php';
require_once __DIR__ . '/feature/TestFeature.php';


class GithubApi2Features
{
    public static function make_feature(string $name)
    {
        switch ($name) {
            case "base":
                return new GithubApi2BaseFeature();
            case "test":
                return new GithubApi2TestFeature();
            default:
                return new GithubApi2BaseFeature();
        }
    }
}

File: py/features.py
# GithubApi2 SDK feature factory

from feature.base_feature import GithubApi2BaseFeature
from feature.test_feature import GithubApi2TestFeature


def _make_feature(name):
    features = {
        "base": lambda: GithubApi2BaseFeature(),
        "test": lambda: GithubApi2TestFeature(),
    }
    factory = features.get(name)
    if factory is not None:
        return factory()
    return features["base"]()
